cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancement/dataset_navigator.spec.js

```diff
@@ -0,0 +1,196 @@
+  cy,
+  Cypress.env('openSearchUrl')
+);
+
+const indexSet = [
+  'logstash-2015.09.22',
+  'logstash-2015.09.21',
+  'logstash-2015.09.20',
+];
+
+describe('dataset navigator', { scrollBehavior: false }, () => {
+  before(() => {
+    CURRENT_TENANT.newTenant = 'global';
+    cy.fleshTenantSettings();
+    cy.deleteAllIndices();
+    cy.deleteSavedObjectByType('index-pattern');
+  });
+
+  describe('empty state', () => {
+    it('no index pattern', function () {
+      // Go to the Discover page
+      miscUtils.visitPage(
+        `app/data-explorer/discover#/?_g=(filters:!(),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))`
+      );
+
+      cy.waitForLoaderNewHeader();
+      cy.getElementByTestId('discoverNoIndexPatterns');
+    });
+  });
+
+  describe('select indices', () => {
+    before(() => {
+      testFixtureHandler.importJSONMapping(
+        'cypress/fixtures/dashboard/opensearch_dashboards/query_enhancement/mappings.json.txt'
+      );
+
+      testFixtureHandler.importJSONDoc(
+        'cypress/fixtures/dashboard/opensearch_dashboards/query_enhancement/data.json.txt'
+      );
+
+      // Go to the Discover page
+      miscUtils.visitPage(`app/data-explorer/discover#/`);
+
+      cy.waitForLoaderNewHeader();
+    });
+
+    it('with SQL as default language', function () {
+      cy.getElementByTestId(`datasetSelectorButton`).click();
+      cy.getElementByTestId(`datasetSelectorAdvancedButton`).click();
+      cy.get(`[title="Indexes"]`).click();
+      cy.get(`[title="Default Cluster"]`).click();
+      cy.get(`[title="timestamp-nanos"]`).click();
+      cy.getElementByTestId('datasetSelectorNext').click();
+
+      cy.get(`[class="euiModalHeader__title"]`).should(
+        'contain',
+        'Step 2: Configure data'
+      );
+      // should have two options: SQL and PPL
+      cy.getElementByTestId('advancedSelectorLanguageSelect')
+        .get('option')
+        .should('have.length', 2);
+
+      //select SQL
+      cy.getElementByTestId('advancedSelectorLanguageSelect').select('SQL');
+      cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+      cy.waitForLoaderNewHeader();
+
+      // Selected language in the language picker should be SQL
+      // bug: SQL won't be selected in cypress; manually click SQL in language selector
+      // cy.getElementByTestId('queryEditorLanguageSelector').should(
+      //   'contain',
+      //   'SQL'
+      // );
+
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`)
+        .should('have.length', 2)
+        .eq(1)
+        .click({
+          force: true,
+        });
+      cy.waitForLoaderNewHeader();
+      cy.get(`[data-test-subj="queryResultCompleteMsg"]`).should('be.visible');
+
+      // Switch language to PPL
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`).eq(0).click({
```

Comment:
nit: i don't think we want to rely on the order. i think the order of the languages is not a specific order on purpose.

Reply:
changed to use name not order